binary_and_decimal.rb

```diff
@@ -14,36 +30,55 @@ def binary_to_decimal
 #       compare individual elements with each other and
 #       you may retrieve the length of an array.
 def array_equals(array1, array2)
-  puts "Not implemented"
-  return true
-end
-
-# --- END OF METHODS ---
+  # I nested lines 36-42 because some of the arrays had different numbers
+  # of elements and wanted to account for the difference.
+  if array1.length == array2.length
+    array1.each_index do |index|
+      if array1.values_at(index) == array2.values_at(index)
+        return true
+      else
+        return false
+      end # if/else
+    end # each_index loop
+  else
+    # I am unsure why the "return puts" code works on line 44. I tried
+    # assigning the 'Make sure ...' string to a variable and then returning
+    # the variable which did not work.
```

Comment:
On line 47, your code ends up returning `nil`. That's what `puts "Make sure your arrays have the same number of elements." evaluates to. It currently works because `nil` is falsy. (See https://github.com/Ada-Developers-Academy/jump-start-live/blob/master/lessons/day2/boolean_expressions.md for more on this.)
When you return a string instead, it ends up evaluating to true since strings are truthy and hence the test case doesn't work.
Whilst the truthy and falsy aspects work a certain way in Ruby, this is not the case with most languages. So, I'd recommend adhering to the specifications and explicitly returning true or false.